# Collection Strategy Analysis

## Case Description

The Credit department has decided to prioritize collections on Contigo A customers, since they have the largest DQ rate. A collections analyst has determined that a DQ rate of 10% will be enough to render this customer segment profitable. The team has decided to apply immediate action from month 1 and is looking for a collection strategy to recover (at least!) 175 of the 675 accounts that went delinquent, out of the starting 5,000 Contigo A accounts.

One idea is to collect by calling the customer. Using the previous collection results, the data science team has created a model to estimate contactability (i.e. the probability of reaching the customer successfully by calling) and has delivered their calculations to you (please, consult the spreadsheet called contactabilty_scores). The collections team estimates that the customer will pay around three quarters of
the time after being contacted, and it has no chance to pay if they never connect the call. Calling the customer is quite expensive! Operations has determined that a phone call campaign costs around 30 pesos per DQ customer, regardless of the contact result.

### Probability Diagram

<img 
    src="https://i.imgur.com/ZKsCGqI.png"
    height=300
    width=650
/>

## Answering Questions

In [19]:
import pandas as pd

In [20]:
# Loads the dataset containing aggregated results
agg_results = pd.read_csv("./collections_results.csv")
# Format numbers to have commas per thousand and two decimals
agg_results = agg_results.applymap(lambda x: "{:,.2f}".format(x))

# Loads the dataset containing non-aggregated results
non_agg_results = pd.read_csv("./collections_campaign.csv")

### Question 1

Assuming that we launch this campaign to all of the 675 DQ customers, a manager asks you to compute its probable result, asking you to deliver estimates on the following KPIs:

- Recovered accounts (accounts we get out of delinquency):

In [22]:
print(f"The total number of accounts recovered was: {agg_results.total_accounts_recovered[0]}")

The total number of accounts recovered was: 212.00


- Recovered total debt:

In [23]:
print(f"The total debt recovered was: {agg_results.total_debt_recovered[0]}")

The total debt recovered was: 212,024.25


- Revenue coming from collected interest:
> **Note**: The total debt collected is not included in revenue, why?

The total debt collected is not included in the revenue because it needs to be paid back to creditors. The revenue comes from any type of interest collected during the period (e.g. ordinary interest and late payment interest).

In [24]:
print(f"The total interest recovered was: {agg_results.total_interest_recovered[0]}")

The total interest recovered was: 31,803.64


- Total campaign cost:

In [26]:
print(f"The total campaign cost was: {agg_results.total_campaign_cost[0]}")

The total campaign cost was: 20,250.00


- DQ losses coming from uncollected total debt:

In [27]:
print(f"The total delinquency losses was: {agg_results.total_uncollected_debt[0]}")

The total delinquency losses was: 462,975.75


- Collections income:

In [29]:
print(f"The total collections income was: {agg_results.collections_income[0]}")

The total collections income was: 223,577.89


### Question 2

What is the impact delivered on overall business performance by this campaign? To answer this question, we provide the results of month 1 without collections in the table below. 

**Table 1**. Performance of Contigo A at month 1.

| **Month** | **Accounts** | **DQ Accounts** | **Interest Revenue** | **DQ Losses** | **Operative Cost** | **Net Income** |
|:---------:|:------------:|:---------------:|:--------------------:|:-------------:|:------------------:|:--------------:|
| 1         | 5,000        | 675             | $648,750.00          | $675,000.00   | $150,000           | -$176,250.00   |

Using the results from the previous question and the data coming from the table, provide a justification for applying or not this collection strategy.

The main results of the collection campaign were the following:

- 21% more accounts recovered than the minimum established by the collection team (212 vs 174).
- Positive net income of $47,327.89 during month 1.

I consider that these results are a valid justification of executing this collection strategy for the first month.

### Question 3

Discuss some possible improvements to this campaign.

To improve this collection strategy, we can utilize multiple communication channels, such as email, mobile app notifications, and SMS, to increase the chances of reaching the customer while reducing costs.

To encourage action, we can inform the customer of the pros and cons of paying their debt. This can be achieved by creating a model that calculates how the current due debt impacts their credit score (including the number of points that will increase or decrease) and how much more credit other financial institutions can offer if the payment is made.

## Non-Aggregated Results

In [30]:
non_agg_results.describe()

customer_id  contactability_score  total_debt  interest  \
count        675.0            675.000000       675.0     675.0   
mean         338.0              0.418813      1000.0     150.0   
std          195.0              0.286737         0.0       0.0   
min            1.0              0.001000      1000.0     150.0   
25%          169.5              0.166500      1000.0     150.0   
50%          338.0              0.378000      1000.0     150.0   
75%          506.5              0.651000      1000.0     150.0   
max          675.0              0.999000      1000.0     150.0   

       phone_call_cost  answer_payment_prob  recov_debt   recov_int  \
count            675.0           675.000000  675.000000  675.000000   
mean              30.0             0.314110  314.110000   47.116500   
std                0.0             0.215052  215.053018   32.257953   
min               30.0             0.000800    0.750000    0.112500   
25%               30.0             0.124850  124.875000   18.731250   
50%               30.0             0.283500  283.500000   42.525000   
75%               30.0             0.488250  488.250000   73.237500   
max               30.0             0.749300  749.250000  112.387500   

       total_recov  
count   675.000000  
mean    361.226500  
std     247.310971  
min       0.862500  
25%     143.606250  
50%     326.025000  
75%     561.487500  
max     861.637500